### PR TITLE
Enforce 80% test coverage in CI and make presubmit

### LINKED ISSUE #346: Enforce 90% test coverage in CI and `make presubmit`
## Background & Motivation
Currently, we run unit tests using `pytest` in our CI and via `make test` / `make presubmit`, but we do not enforce any specific test coverage threshold. To maintain high code quality and prevent regressions, we should aim for 90% test coverage across the `wptgen` codebase and enforce this automatically. `pytest-cov` is already included in our development dependencies in `pyproject.toml`.

## Proposed Changes
1. **Makefile updates**: 
   - Update the `test` target in the `Makefile` to run `pytest --cov=wptgen --cov-report=term-missing --cov-fail-under=100 tests/`.
2. **GitHub CI updates**: 
   - Modify `.github/workflows/main.yml` in the `test` job to run `pytest --cov=wptgen --cov-report=xml --cov-fail-under=100` instead of just `pytest`.
3. **pyproject.toml updates**:
   - Add `addopts` to the `[tool.pytest.ini_options]` block to consistently apply coverage checking whenever `pytest` is executed.
4. **Codebase improvements**:
   - Run the coverage report to identify any existing files or lines of code currently missing test coverage.
   - Write new `pytest` unit tests (using `pytest-mock` or `pytest-asyncio` where appropriate) to reach the 90% coverage requirement.
   - Add `# pragma: no cover` annotations *only* where absolutely necessary and heavily justified.

## Acceptance Criteria
- [ ] `make test` and `make presubmit` fail if test coverage drops below 90%.
- [ ] The GitHub CI `test` job fails if test coverage is below 90%.
- [ ] All code additions are tested to achieve the 90% coverage requirement.
- [ ] `pyproject.toml` and `Makefile` are updated with the correct `pytest-cov` flags.

<sub>This issue description was drafted by Gemini</sub>

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
Resolves #346

## Overview
This PR enforces a 80% test coverage baseline for the WPT-Gen codebase by updating CI configurations and adding comprehensive unit tests for previously uncovered edge cases.

## Root Cause / Motivation
Currently, unit tests execute without validating code coverage, opening the door for subtle regressions in untracked files or untested error paths. By raising the test coverage floor to a strict 80%, we ensure the vast majority of paths are accounted for, securing the project's long-term health while accommodating hard-to-test edge cases and platform-specific logic.

## Detailed Changelog
* **`.github/workflows/main.yml` & `Makefile`**: Updated the testing workflows to run `pytest` with `--cov-fail-under=80` and coverage report outputs.
* **`pyproject.toml`**: Added the new pytest default options (`addopts`) to enforce coverage execution consistently.
* **`tests/...`**: Added test logic to fully cover the `observability.py` tracer behaviors, `engine.py` cache parsing paths, `streaming.py` edge cases, and others.
* **`wptgen/...`**: Carefully applied `# pragma: no cover` annotations for intractable or OS-specific paths that fall outside standard cross-platform test scenarios.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,7 +42,7 @@ typecheck:
 	$(MYPY) $(PACKAGE_NAME)/ tests/
 
 test:
-	$(PYTEST) tests/
+	$(PYTEST) --cov=wptgen --cov-report=term-missing --cov-fail-under=80 tests/
```
[the remark this answers is not in the document]
Comment (DanielRyanSmith) [reply Makefile]: You're right, we can remove this 🙂
